src/Microsoft.Extensions.Configuration.Json/JsonConfigurationProvider.cs

```diff
@@ -52,18 +52,20 @@ public override void Load(Stream stream)
 
         private static string RetrieveErrorContext(JsonReaderException e, IEnumerable<string> fileContent)
         {
-            string errorLine;
+            string errorLine = null;
             if (e.LineNumber >= 2)
             {
                 var errorContext = fileContent.Skip(e.LineNumber - 2).Take(2).ToList();
-                errorLine = errorContext[0].Trim() + Environment.NewLine + errorContext[1].Trim();
+                if (errorContext.Count() >= 2)
```

Comment:
The specific case that was null refing is with really short json like the test I added, the error line number would be 5, and we only had 3 lines of input, and errorContext would returns a single element of null, so everything blows up with a null ref (errorContext[0].Trim = boom) since there was no checking going on.  The fix just adds a guard against that error condition, and I made the other original fall thru behavior get executed now (which results in an empty error line)